fraos/src/seqno.rs

```diff
@@ -114,12 +142,47 @@ impl SeqNoIndex {
     pub(crate) fn mmaps_count(&self) -> Result<usize, FraosError> {
         self.inner.mmaps_count()
     }
+
+    pub(crate) fn find_actual_end(&self) -> Result<usize, FraosError> {
+        let mut start = 0;
+        let len = self.len();
+        let mut end = self.len();
+
+        // empty index was created or index is empty
+        if self.get_length_at(start)? == 0 || end == 0 {
+            return Ok(0);
+        }
+
+        // all elements are non-zero
+        if self.get_length_at(end.saturating_sub(1))? != 0 {
+            return Ok(end);
+        }
+
+        while start < len.saturating_sub(1) {
+            if self.get_length_at(start)? != 0 && self.get_length_at(start + 1)? == 0 {
```

Comment:
can `self.get_length_at(start)? != 0` ever be false? in the first iteration it has to be false otherwise the index was empty and there is the early return. In the other case, it's the `mid` of the previous iteration, in which case it's already checked that it was not zero  because of the if below.

Reply:
good catch